sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -385,8 +385,10 @@ object DateTimeUtils {
    * microseconds where microsecond 0 is 1970-01-01 00:00:00Z.
    */
   def instantToMicros(instant: Instant): Long = {
-    val us = Math.multiplyExact(instant.getEpochSecond, MICROS_PER_SECOND)
-    val result = Math.addExact(us, NANOSECONDS.toMicros(instant.getNano))
+    // we don't need to take care overflow here, since if overflow occurred,
```

Comment:
I don't think this is the right fix. The range of `instant.getEpochSecond` is  [-31557014167219200L, 31556889864403199L]

Reply:
Yes, I've found that too, so I've made a fixed condition for the Long.minValue, could you re-review that for me. Thanks